### ISSUE TITLE
关于微调的一些问题

### ISSUE THREAD

**Issue by Jeremy-J-J** [Francis-Rings/StableAnimator#60]
请问一下大佬关于微调的一些问题：
(1) 关于微调的验证集 --validation_image_folder(ground_truth)、--validation_control_folder(poses)、--validation_image 分别放的是什么？我的理解是-validation_image_folder(ground_truth)放的可以随便一个短视频保存的逐帧(不用缩放到512x512或576x1024)，--validation_control_folder(poses)是放对应-validation_image_folder(ground_truth)提取的关键点，然后--validation_image是这个视频的第一帧的图片？还是另外的参考图？
(2) 关于微调的epochs，看论文里说epochs=20，但comand_finetune.sh中的默认epochs=60000，这里就有两个问题，一是这个epoch=6000似乎太大，我在训练集rec和vec分别只有1个情况下去做微调 (4卡A800)，都需要280h；另一个问题是comand_finetune.sh和comand_train.sh中的epoch都是6000，似乎不太合理。所以想问一下大佬，对于微调场景，比如微调数据集只有1~2个视频数据的时候，微调控参怎么配置可以比较快速获得一个好的生成效果。

下面是我试验了epochs=20的微调效果，效果看起来很不理想：

![gif-2024-12-28 at 14 57 45](https://github.com/user-attachments/assets/cf3ac2fb-bde2-4441-a59b-aeab31775222)

我的  command_finetune.sh 的配置如下：
```
CUDA_VISIBLE_DEVICES=3,2,1,0 accelerate launch train.py \
 --pretrained_model_name_or_path="/home/StableAnimator/checkpoints/stable-video-diffusion-img2vid-xt" \
 --finetune_mode=True \
 --posenet_model_finetune_path="/home/StableAnimator/checkpoints/Animation/pose_net.pth" \
 --face_encoder_finetune_path="/home/StableAnimator/checkpoints/Animation/face_encoder.pth" \
 --unet_model_finetune_path="/home/StableAnimator/checkpoints/Animation/unet.pth" \
 --output_dir="/home/StableAnimator/checkpoints/Animation" \
 --data_root_path="/home/StableAnimator/animation_data" \
 --rec_data_path="/home/StableAnimator/animation_data/video_rec_path.txt" \
 --vec_data_path="/home/StableAnimator/animation_data/video_vec_path.txt" \
 --validation_image_folder="/home/StableAnimator/validation/ground_truth" \
 --validation_control_folder="/home/StableAnimator/validation/poses" \
 --validation_image="/home/StableAnimator/validation/ground_truth/frame_0.png" \
 --num_workers=8 \
 --lr_warmup_steps=10 \
 --sample_n_frames=16 \
 --learning_rate=1e-5 \
 --per_gpu_batch_size=1 \
 --num_train_epochs=20 \
 --mixed_precision="fp16" \
 --gradient_accumulation_steps=1 \
 --checkpointing_steps=10 \
 --validation_steps=50 \
 --gradient_checkpointing \
 --checkpoints_total_limit=50 \
#  --resume_from_checkpoint="latest"
```

**Comment by Francis-Rings** [comment]
Hi, regarding your first question: --validation_image, --validation_image_folder, and --validation_control_folder refer to a random frame selected from the video clip, a random continuous subclip from the same video, and the corresponding human skeleton images extracted from the --validation_image_folder, respectively. The images stored in the validation directory should have a resolution of either 512x512 or 576x1024.

As for your second question: I set the number of epochs to 60,000 for the convenience of observing intermediate results. The convergence speed depends on the scale and quality of your training datasets. For determining the optimal training steps, I recommend monitoring the intermediate results and terminating the training process when the model reaches peak performance.

**Comment by Jeremy-J-J** [comment]
Thank you for your reply.
I have another question.If my fine-tuning dataset contains only one video, do I need to create both rec and vec based on this single video, or is it sufficient to make just one? Additionally, if my fine-tuning dataset contains only one video, can the validation set be the video itself?

**Comment by Francis-Rings** [comment]
If you tend to finetune StableAnimator on a single video, there's no need to create rec and vec, as they are intended for mixed-resolution parallel training.  From my perspective, the more videos you use for training, the better the performance of StableAnimator will be. Additionally, when your dataset contains only one video, the video itself can serve as the validation set.

**Comment by aleeyang** [comment]
@Francis-Rings Hi, I have a question: what do 'rec' and 'vec' specifically refer to? If I have a multi-person video dataset, how can I create these two files? 
Thanks.

**Comment by Francis-Rings** [comment]
> @Francis-Rings Hi, I have a question: what do 'rec' and 'vec' specifically refer to? If I have a multi-person video dataset, how can I create these two files? Thanks.

Hi, `rec` and `vec` refer to 512x512 video clips and 576x1024 video clips, respectively. `rec` and `vec` are related to the resolution of your training videos instead of protagonist numbers of your training videos. For more details, please refer to the training tutorial in the README file.

**Comment by Jeremy-J-J** [comment]
> If you tend to finetune StableAnimator on a single video, there's no need to create rec and vec, as they are intended for mixed-resolution parallel training. From my perspective, the more videos you use for training, the better the performance of StableAnimator will be. Additionally, when your dataset contains only one video, the video itself can serve as the validation set.

If I don't create rec and vec files, how should I configure the --rec_data_path and --vec_data_path in the fine-tuning script command_finetune.sh, and how do I pass the paths of the video frames and pose data to be fine-tuned to the script?

**Comment by Francis-Rings** [comment]
If you want to train StableAnimator on a single resolution, you can use the following command:
```
bash command_train_single.sh
```
You can customize the resolution by modifying `--dataset_width` and `--dataset_height`, both of which default to 512. I have updated the training tutorial. For more details about single resolution setting, please refer to the the training tutorial in the README file.

**Comment by Jeremy-J-J** [comment]
> If you want to train StableAnimator on a single resolution, you can use the following command:
> 
> ```
> bash command_train_single.sh
> ```
> 
> You can customize the resolution by modifying `--dataset_width` and `--dataset_height`, both of which default to 512. I have updated the training tutorial. For more details about single resolution setting, please refer to the the training tutorial in the README file.

感谢大佬
